src/distPlus/distribution/Distributions.re

```diff
@@ -366,6 +407,41 @@ module Mixed = {
           discreteProbabilityMassFraction,
         };
       };
+
+      let getMean = (t: t): float => {
+        let discreteProbabilityMassFraction =
+          t.discreteProbabilityMassFraction;
+        switch (discreteProbabilityMassFraction) {
+        | 1.0 => Discrete.T.getMean(t.discrete)
+        | 0.0 => Continuous.T.getMean(t.continuous)
+        | _ =>
+          Discrete.T.getMean(t.discrete)
+          *. discreteProbabilityMassFraction
+          +. Continuous.T.getMean(t.continuous)
+          *. (1.0 -. discreteProbabilityMassFraction)
+        };
+      };
+
+      let getVariance = (t: t): float => {
+        let discreteProbabilityMassFraction =
+          t.discreteProbabilityMassFraction;
+        let getMeanOfSquares = (t: t) => {
+          Discrete.T.getMean(XYShape.Analysis.squareXYShape(t.discrete))
+          *. t.discreteProbabilityMassFraction
+          +. XYShape.Analysis.getMeanOfSquaresContinuousShape(t.continuous)
+          *. (1.0 -. t.discreteProbabilityMassFraction);
+        };
+        switch (discreteProbabilityMassFraction) {
+        | 1.0 => Discrete.T.getVariance(t.discrete)
+        | 0.0 => Continuous.T.getVariance(t.continuous)
```

Comment:
I don't think these lines should be necessary. Mixed() shapes should really only be in cases where there is significant prob. mass on both.